src/core/geometry/geom3/reTesselateCoplanarPolygons.js

```diff
@@ -7,332 +7,334 @@ const poly3 = require('../poly3')
-            activepolygon.topright = vertices2d[newrightvertexindex]
-            let nextleftvertexindex = newleftvertexindex + 1
-            if (nextleftvertexindex >= numvertices) nextleftvertexindex = 0
-            activepolygon.bottomleft = vertices2d[nextleftvertexindex]
-            let nextrightvertexindex = newrightvertexindex - 1
-            if (nextrightvertexindex < 0) nextrightvertexindex = numvertices - 1
-            activepolygon.bottomright = vertices2d[nextrightvertexindex]
-          }
-        } // if polygon has corner here
-      } // for activepolygonindex
-      let nextycoordinate
-      if (yindex >= ycoordinates.length - 1) {
-        // last row, all polygons must be finished here:
-        activepolygons = []
-        nextycoordinate = null
-      } else // yindex < ycoordinates.length-1
-      {
-        nextycoordinate = Number(ycoordinates[yindex + 1])
-        let middleycoordinate = 0.5 * (ycoordinate + nextycoordinate)
-        // update activepolygons by adding any polygons that start here:
-        let startingpolygonindexes = topy2polygonindexes[ycoordinate_as_string]
-        for (let polygonindex_key in startingpolygonindexes) {
-          let polygonindex = startingpolygonindexes[polygonindex_key]
-          let vertices2d = polygonvertices2d[polygonindex]
-          let numvertices = vertices2d.length
-          let topvertexindex = polygontopvertexindexes[polygonindex]
-          // the top of the polygon may be a horizontal line. In that case topvertexindex can point to any point on this line.
-          // Find the left and right topmost vertices which have the current y coordinate:
-          let topleftvertexindex = topvertexindex
-          while (true) {
-            let i = topleftvertexindex + 1
-            if (i >= numvertices) i = 0
-            if (vertices2d[i].y !== ycoordinate) break
-            if (i === topvertexindex) break // should not happen, but just to prevent endless loops
-            topleftvertexindex = i
-          }
-          let toprightvertexindex = topvertexindex
-          while (true) {
-            let i = toprightvertexindex - 1
-            if (i < 0) i = numvertices - 1
-            if (vertices2d[i].y !== ycoordinate) break
-            if (i === topleftvertexindex) break // should not happen, but just to prevent endless loops
-            toprightvertexindex = i
-          }
-          let nextleftvertexindex = topleftvertexindex + 1
+    // update activepolygons for this y coordinate:
+    // - Remove any polygons that end at this y coordinate
+    // - update leftvertexindex and rightvertexindex (which point to the current vertex index
+    //   at the the left and right side of the polygon
+    // Iterate over all polygons that have a corner at this y coordinate:
+    const polygonindexeswithcorner = ycoordinatetopolygonindexes[ycoordinateasstring]
+    for (let activepolygonindex = 0; activepolygonindex < activepolygons.length; ++activepolygonindex) {
```

Comment:
I seem to recall that extracting `activepolygons.length` to a variable before the loop is good optimising method